app/controllers/cities_controller.rb

```diff
@@ -0,0 +1,66 @@
+class CitiesController < ApplicationController
+  before_action :set_city, only: %i[show edit update destroy]
+
+  # GET /cities or /cities.json
+  def index
+    @cities = City.all
+
+    render json: @cities
+  end
+
+  # GET /cities/1 or /cities/1.json
+  def show; end
+
+  # GET /cities/new
+  def new
+    @city = City.new
+  end
+
+  # GET /cities/1/edit
+  def edit; end
+
+  # POST /cities or /cities.json
+  def create
+    @city = City.new(city_params)
+    if @city.save
+      render json: @city, status: :created, location: @city
+    else
+      render json: @city.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /cities/1 or /cities/1.json
+  def update
+    respond_to do |format|
+      if @city.update(city_params)
+        format.html { redirect_to city_url(@city), notice: 'City was successfully updated.' }
+        format.json { render :show, status: :ok, location: @city }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @city.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /cities/1 or /cities/1.json
+  def destroy
+    @city.destroy
+
+    respond_to do |format|
+      format.html { redirect_to cities_url, notice: 'City was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
```

Comment:
- [ ] What happens if the city is not successfully destroyed?  Kindly, add some error handling like you did on the update method (line 33). 👍